src/components/CustomButton.vue

```diff
@@ -0,0 +1,108 @@
+<script setup>
+import { defineProps, defineEmits } from "vue";
+const props = defineProps({
+  iconSource: {
+    type: String,
+    default: "",
+  },
+  iconAlternativeText: {
+    type: String,
+    default: "",
+  },
+  label: {
+    type: String,
+    default: "",
+  },
+  disabled: {
+    type: Boolean,
+    default: false,
+  },
+});
+
+defineEmits({
+  click: {},
+});
+
+function showIcon() {
+  return props.iconSource || props.iconAlternativeText;
+}
+
+function showLabel() {
+  return props.label;
+}
+
+function isDisabled() {
+  return props.disabled;
+}
+</script>
+
+<template>
+  <button
+    v-if="isDisabled"
+    :disabled="isDisabled()"
+    :class="[
+      $style.button,
+      $style['button--default'],
+      { [$style['button--disabled']]: isDisabled() },
+    ]"
+    type="button"
+    @click.stop="$emit('click')"
+  >
+    <img
+      v-if="showIcon()"
+      :class="$style['button__icon']"
+      :src="iconSource"
+      :alt="iconAlternativeText"
+    />
+    <span v-if="showLabel()" :class="$style['button__label']">{{ label }}</span>
+  </button>
+</template>
+
+<style module lang="scss">
+.button {
+  display: flex;
+  justify-content: center;
+  align-items: center;
+  width: 100%;
+  height: 40px;
+  margin: 0 20px 20px 0;
+
+  &__icon {
+    object-fit: cover;
+    width: 30px;
+    height: 30px;
+    gap: 20px;
+  }
+
+  &__label {
+    font-size: 1rem;
+    margin: 5px 10px 5px 10px;
```

Comment:
```suggestion
    margin: 5px 10px;
```

You can always define these vars in clock order. See:

- top -> right -> bottom -> left: `margin: 5px 5px 5px 5px;`
- top / bottom -> right / left: `margin: 5px 5px;`
- top / right / bottom / left: `margin: 5px;`